fe/fe-core/src/main/java/org/apache/doris/datasource/HMSExternalDataSource.java

```diff
@@ -32,31 +48,170 @@ public class HMSExternalDataSource extends ExternalDataSource {
     private static final Logger LOG = LogManager.getLogger(HMSExternalDataSource.class);
     public static final String HIVE_METASTORE_URIS = "hive.metastore.uris";
 
-    public HMSExternalDataSource() {
+    protected final String hiveMetastoreUris;
+    protected final HiveMetaStoreClient client;
+    private final HMSType hmsType;
+
+    private ConcurrentHashMap<String, DatabaseIf> nameToDb = new ConcurrentHashMap();
+
+    /**
+     * Hive metastore data source implementation.
+     *
+     * @param hiveMetastoreUris e.g. thrift://127.0.0.1:9083
+     * @param hmsType Hive metastore type. Hive, Iceberg or Hudi.
+     */
+    public HMSExternalDataSource(String hiveMetastoreUris, HMSType hmsType) throws DdlException {
+        this.hiveMetastoreUris = hiveMetastoreUris;
+        this.hmsType = hmsType;
+        HiveConf hiveConf = new HiveConf();
+        hiveConf.setVar(HiveConf.ConfVars.METASTOREURIS, hiveMetastoreUris);
+        try {
+            client = new HiveMetaStoreClient(hiveConf);
+        } catch (MetaException e) {
+            LOG.warn("Create HiveMetaStoreClient failed: {}", e.getMessage());
+            throw new DdlException("Create HiveMetaStoreClient failed: " + e.getMessage());
+        }
+    }
+
+    public HMSType getHmsType() {
+        return hmsType;
     }
 
     @Override
     public List<String> listDatabaseNames(SessionContext ctx) {
+        try {
```

Comment:
I think we should throw exception instead of return null here?
Better change the interface.